device/tt_silicon_driver.cpp

```diff
@@ -1197,216 +1083,29 @@ void tt_SiliconDevice::set_fallback_tlb_ordering_mode(const std::string& fallbac
 void tt_SiliconDevice::init_pcie_iatus() {
     int num_enabled_devices = m_pci_device_map.size();
     log_debug(LogSiliconDriver, "tt_SiliconDevice::init_pcie_iatus() num_enabled_devices: {}", num_enabled_devices);
-    log_assert(m_num_host_mem_channels <= g_MAX_HOST_MEM_CHANNELS, "Maximum of {} 1GB Host memory channels supported.",  g_MAX_HOST_MEM_CHANNELS);
 
     for (auto &src_device_it : m_pci_device_map){
         int logical_id = src_device_it.first;
         PCIDevice* src_pci_device = src_device_it.second.get();
 
         // Device to Host (multiple channels)
-        for (int channel_id = 0; channel_id < m_num_host_mem_channels; channel_id++) {
-            if (hugepage_mapping.at(logical_id).at(channel_id)) {
-                std::uint32_t region_size = HUGEPAGE_REGION_SIZE;
+        for (int channel_id = 0; channel_id < src_pci_device->get_num_host_mem_channels(); channel_id++) {
+            hugepage_mapping hugepage_map = src_pci_device->get_hugepage_mapping(channel_id);
+            if (hugepage_map.mapping) {
+                std::uint32_t region_size = hugepage_map.mapping_size;
                 if (channel_id == 3) region_size = 805306368; // Remove 256MB from full 1GB for channel 3 (iATU limitation)
```

Comment:
can you add {} around this if, and also put this number `805306368` in a constant variable